pkg/porter/editor.go

```diff
@@ -0,0 +1,86 @@
+package porter
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"get.porter.sh/porter/pkg/context"
+)
+
+const (
+	defaultEditor        = "vi"
+	defaultEditorWindows = "notepad"
+)
+
+// Editor displays content to a user using an external text editor, like vi or notepad.
+// The content is captured and returned.
+//
+// The `EDITOR` environment variable is checked to find an editor.
+// Failing that, use some sensible default depending on the operating system.
+//
+// This is useful for editing things like configuration files, especially those
+// that might be stored on a remote server. For example: the content could be retrieved
+// from the remote store, edited locally, then saved back.
+type Editor struct {
+	Data     []byte
+	Filename string
+	Context  *context.Context
+}
+
+func editorArgs() []string {
+	editor := defaultEditor
+	if runtime.GOOS == "windows" {
+		editor = defaultEditorWindows
+	}
+	if os.Getenv("EDITOR") != "" {
+		editor = os.Getenv("EDITOR")
+	}
+
+	args := []string{}
+	if strings.ContainsAny(editor, "\"'\\") {
```

Comment:
Can you change this to a raw string so that it's easier to tell what characters you are testing for? A comment explaining what is going on with an example of the args and why you are looking for those characters and then splitting on space would really help too.

I think you are looking for `\`, `"` and `'`
```suggestion
	if strings.ContainsAny(editor, `"'\`) {
```

Reply:
Will do.